src/app/wallets/update-on-chain-receipt.ts

```diff
@@ -249,7 +244,7 @@ const processTxForWallet = async (
             amount: Number(
               minorToMajorUnit({
                 amount: creditAccountAdditionalMetadata.displayAmount,
-                displayMajorExponent: MajorExponent.STANDARD,
+                displayCurrency: creditAccountAdditionalMetadata.displayCurrency,
```

Comment:
Like here. I think after my refactor (after this branch) the `displayCurrency` could come directly from `amount` field if it's typed as `DisplayAmount` and then we could remove the `displayCurrency` param here